apps/taxes/src/modules/taxes/active-tax-provider.ts

```diff
@@ -1,45 +1,113 @@
-import { TaxBaseFragment } from "../../../generated/graphql";
+import {
+  MetadataItem,
+  OrderCreatedSubscriptionFragment,
+  OrderFulfilledSubscriptionFragment,
+  TaxBaseFragment,
+} from "../../../generated/graphql";
 import { createLogger } from "../../lib/logger";
 import { ChannelConfig } from "../channels-configuration/channels-config";
 import { ProviderConfig } from "../providers-configuration/providers-config";
-import { AvataxProvider } from "../avatax/avatax-provider";
-import { TaxJarProvider } from "../taxjar/taxjar-provider";
-import { TaxProvider } from "./tax-provider";
+import { AvataxWebhookService } from "../avatax/avatax-webhook.service";
+import { TaxJarWebhookService } from "../taxjar/taxjar-webhook.service";
+import { ProviderWebhookService } from "./tax-provider-webhook";
 import { TaxProviderError } from "./tax-provider-error";
 import pino from "pino";
+import { getAppConfig } from "../app-configuration/get-app-config";
 
-export class ActiveTaxProvider {
-  private client: TaxProvider;
+type ActiveTaxProviderResult =
+  | { ok: true; data: ActiveTaxProvider }
+  | { ok: false; data: null; error: string };
+
+export function getActiveTaxProvider(
+  channelSlug: string | undefined,
+  metadata: MetadataItem[]
+): ActiveTaxProviderResult {
+  const logger = createLogger({ service: "getActiveTaxProvider" });
+
+  if (!channelSlug) {
+    logger.error("Channel slug is missing");
+    return { error: "Channel slug is missing", data: null, ok: false };
+  }
+
+  if (!metadata.length) {
+    logger.error("App metadata is missing");
+    return { error: "App metadata is missing", data: null, ok: false };
+  }
+
+  const { providers, channels } = getAppConfig(metadata);
+
+  const channelConfig = channels[channelSlug];
+
+  if (!channelConfig) {
+    // * will happen when `order-created` webhook is triggered by creating an order in a channel that doesn't use the tax app
+    logger.info(`Channel config not found for channel ${channelSlug}`);
+    return { error: `Channel config not found for channel ${channelSlug}`, data: null, ok: false };
+  }
+
+  const providerInstance = providers.find(
```

Comment:
Im confused with the naming. `providerInstance` for me sounds like instance of the class. This is providerConfig or something like this.

When I see `new ActiveTaxProvider(providerInstance, channelConfig);` I assume you inject constructed provider into ActiveTaxProvider.